pkg/source/adapter/controlled_adapter.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kafka
```

Comment:
Can we move this to a different package? I'm hoping that control-protocol does not get linked when building the mt source adapter, just to avoid unnecessary informers.

Reply:
I would love to avoid to do that in this PR, because it requires further refactorings (we need to make most of the elements of the adapter public).

> just to avoid unnecessary informers.

In control-protocol only the ctrlreconciler package includes the k8s related stuff, the rest is pure golang, hence the `adapter` package won't depend on k8s, not even transitively.